softvis3d-frontend/src/classes/status/StatusActionQueue.ts

```diff
@@ -21,7 +22,13 @@ export default class StatusActionQueue<T extends StatusAction> {
         console.error("Could not remove action: " + JSON.stringify(action));
     }
 
-    public copyAndUpdate(action: T): StatusActionQueue<T> {
+    /**
+     * As only the content of the action changes, queue has to be recreated
+     * for the @observable to fire an event.
+     *
+     * @returns new instance of StatusActionQueue<T>.
+     */
+    public update(action: T): StatusActionQueue<T> {
```

Comment:
I'm open to everything here. If you have an idea, give it a try. But i think this should be done in a different pull request or event an different issue ticket then.